src/components/shared/ContractFunction.vue

```diff
@@ -183,29 +191,63 @@ const makeBytes32 = (value: string, type: string) => {
   return value
 }
 
+if (props.dataDecoder) {
+  watch(
+    () => reactiveData.items?.map(({ type }) => type),
+    async () => {
+      const value = props.data
+      if (value) {
+        try {
+          const { eth } = getWeb3()
+          const output = eth.abi.decodeParameters(
+            reactiveData.items.map(({ type }) => type),
+            value
+          )
+          for (const [index, item] of Object.entries(output)) {
+            if (Number.isNaN(Number(index))) {
+              continue
+            }
+            reactiveData.items[Number(index)].value = item
+          }
+          emits('update:modelValue', reactiveData.items)
+        } catch (err) {
+          // Ignore
+        }
+      }
+    }
+  )
+}
+
 const output = computed<{ error: undefined | string; value: string }>(() => {
   try {
-    if (!reactiveData.call) {
+    if (!reactiveData.call && !props.dataDecoder) {
       return { value: '0x', error: undefined }
     }
     const { eth } = getWeb3()
-    const callSig = `${reactiveData.call}(${reactiveData.items
-      .map(({ type }) => type)
-      .join(',')})`
+    const callSig = props.dataDecoder
+      ? `${reactiveData.items.map(({ type }) => type).join(',')}`
+      : `${reactiveData.call}(${reactiveData.items
+          .map(({ type }) => type)
+          .join(',')})`
+    const types = reactiveData.items.map(({ type }) => type)
+    const args = reactiveData.items.map(({ value, type, isWei }) => {
+      const makeItem = (value: any) =>
+        /^bytes32/.test(type)
+          ? makeBytes32(value, type) ?? '0x'
+          : makeValue(value, isWei) || ''
+      if (/\[\]$/.test(type)) {
```

Comment:
We may also potentially have fixed size arrays:
```suggestion
      if (/\[(\d*)?\]$/.test(type)) {
```

Reply:
Okay, I'll open a task so we don't forget.

Reply:
https://app.clickup.com/t/86by5pnv8